### ISSUE TITLE
GLB SRAM Should Have Separated Write and Read Ports

### ISSUE THREAD

**Issue by SingularityKChen** [SingularityKChen/dl_accelerator#8]
GLB SRAM should have separated write and read ports because when accumulating PSum, PE Array will read current PSum and maybe before reading finish, it will write back PSum.

**Comment by SingularityKChen** [comment]
Don't know why `readIdxIncReg` will be changed when writing finishes.
So I add enable wire risky to pass the test.

Before:

```scala
  readIdxIncReg := Mux(readDoneWire, 0.U, Mux(nextValidReg, 
    readIdxIncReg + 1.U, readIdxIncReg)
  )
```

After:

```scala
readIdxIncReg := Mux(readDoneWire || !io.ctrlPath.readIO.enable, 0.U,
    Mux(nextValidReg, readIdxIncReg + 1.U, readIdxIncReg)
  )
```

**Comment by SingularityKChen** [comment]
Fix this bug with commit [86acf95](https://github.com/SingularityKChen/dl_accelerator/commit/86acf953b42c8d408b60ecb01240e0caa3fb6687)
